File: app/src/main/java/com/botijasoftware/utils/TexturedQuad.java
package com.botijasoftware.utils;

public class TexturedQuad extends Sprite {

	public TexturedQuad(Vector2 position, Texture texture) {
		super(new Rectanglef(position.X, position.Y, texture.getWidth(), texture.getHeight()), texture);
	}
	
	
	public TexturedQuad(Vector2 position, Texture texture, boolean flip) {
		super(new Rectanglef(position.X, position.Y, texture.getWidth(), texture.getHeight()),texture, 1.0f, 0.0f, new ColorRGBAb((byte)255,(byte)255,(byte)255,(byte)255), flip);
	}
	
	public TexturedQuad(Vector2 position, Texture texture, ColorRGBAb color) {
		super(new Rectanglef(position.X, position.Y, texture.getWidth(), texture.getHeight()),texture, 1.0f, 0.0f, color, false);
	}
	
	public TexturedQuad(Vector2 position, Texture texture, float scale) {
		super(new Rectanglef(position.X, position.Y, texture.getWidth(), texture.getHeight()),texture, scale, 0.0f, new ColorRGBAb((byte)255,(byte)255,(byte)255,(byte)255), false);
	}
	
	public TexturedQuad(Vector2 position, Texture texture, float scale, float rotation) {
		super(new Rectanglef(position.X, position.Y, texture.getWidth(), texture.getHeight()),texture, scale, rotation, new ColorRGBAb((byte)255,(byte)255,(byte)255,(byte)255), false);
	}
	
	public TexturedQuad(Vector2 position, Texture texture, float scale, float rotation, boolean flip) {
		super(new Rectanglef(position.X, position.Y, texture.getWidth(), texture.getHeight()),texture, scale, rotation, new ColorRGBAb((byte)255,(byte)255,(byte)255,(byte)255), flip);
	}
	
	@Override
	public void setTexture(Texture texture) {
		mRectangle.setSize(texture.getWidth(), texture.getHeight());
		super.setTexture(texture);
	}	
	
	
	@Override
	public void setRectangle(Rectanglef rectangle) {
		
		mRectangle.setPosition(rectangle.X, rectangle.Y);
		super.setRectangle(mRectangle);
				
	}
	
	@Override
	public void setSize(float nw, float nh) {
	}
	
	@Override
	public void setSize(Vector2 v) {
	
	}
	
}

File: app/src/main/java/com/botijasoftware/utils/TexturedQuad.kt
package com.botijasoftware.utils

class TexturedQuad : Sprite {

    constructor(position: Vector2, texture: Texture) : super(Rectanglef(position.X, position.Y, texture.width.toFloat(), texture.height.toFloat()), texture)


    constructor(position: Vector2, texture: Texture, flip: Boolean) : super(Rectanglef(position.X, position.Y, texture.width.toFloat(), texture.height.toFloat()), texture, 1.0f, 0.0f, ColorRGBAb(255.toByte(), 255.toByte(), 255.toByte(), 255.toByte()), flip)

    constructor(position: Vector2, texture: Texture, color: ColorRGBAb) : super(Rectanglef(position.X, position.Y, texture.width.toFloat(), texture.height.toFloat()), texture, 1.0f, 0.0f, color, false)

    constructor(position: Vector2, texture: Texture, scale: Float) : super(Rectanglef(position.X, position.Y, texture.width.toFloat(), texture.height.toFloat()), texture, scale, 0.0f, ColorRGBAb(255.toByte(), 255.toByte(), 255.toByte(), 255.toByte()), false)

    constructor(position: Vector2, texture: Texture, scale: Float, rotation: Float) : super(Rectanglef(position.X, position.Y, texture.width.toFloat(), texture.height.toFloat()), texture, scale, rotation, ColorRGBAb(255.toByte(), 255.toByte(), 255.toByte(), 255.toByte()), false)

    constructor(position: Vector2, texture: Texture, scale: Float, rotation: Float, flip: Boolean) : super(Rectanglef(position.X, position.Y, texture.width.toFloat(), texture.height.toFloat()), texture, scale, rotation, ColorRGBAb(255.toByte(), 255.toByte(), 255.toByte(), 255.toByte()), flip)

    override var texture: Texture
        get() = super.texture
        set(texture) {
            mRectangle.setSize(texture.width.toFloat(), texture.height.toFloat())
            super.texture = texture
        }


    override var rectangle: Rectanglef
        get() = super.rectangle
        set(rectangle) {

            mRectangle.setPosition(rectangle.X, rectangle.Y)
            super.rectangle = mRectangle

        }

    override fun setSize(nw: Float, nh: Float) {}

    override fun setSize(v: Vector2) {

    }

}